firmware/arduino/chessboard/chessboard.ino

```diff
@@ -6,6 +6,45 @@
 #define RX2 16
 #define TX2 17
 
+// UART input and flags
+char moveCount;
+volatile char currentState = '0';
```

Comment:
Can you please put this after the definition of the `ArduinoState` enum? And then initialize it as `volatile char currentState = IDLE;`